packages/core/test/Synchronizer.ts

```diff
@@ -0,0 +1,198 @@
+
+let synchronizer: UnirepSocialSynchronizer
+const attestingFee = BigNumber.from(1)
+
+describe('Synchronzier processes events', function () {
+    this.timeout(0)
+
+    let unirepContract
+    let unirepSocialContract: UnirepSocial
+
+    before(async () => {
+        const accounts = await ethers.getSigners()
+        unirepContract = await deployUnirep(accounts[0], {
+            attestingFee,
+        })
+        unirepSocialContract = await deployUnirepSocial(
+            accounts[0],
+            unirepContract.address,
+            {
+                airdropReputation: 30,
+            }
+        )
+        const db = await SQLiteConnector.create(schema, ':memory:')
+        synchronizer = new UnirepSocialSynchronizer(
+            db,
+            defaultProver,
+            unirepContract,
+            unirepSocialContract
+        )
+        // now create an attester
+        await unirepContract
+            .connect(accounts[1])
+            .attesterSignUp()
+            .then((t) => t.wait())
+        await synchronizer.start()
+    })
+
+    afterEach(async () => {
+        await synchronizer.waitForSync()
+        const state = await genUserState(
+            synchronizer.unirepContract.provider,
+            synchronizer.unirepContract.address,
+            new ZkIdentity()
+        )
+        await state.stop()
+    })
+
+    it('submit post should succeed', async () => {
+        const attesterId = BigInt(
+            await unirepContract.attesters(unirepSocialContract.address)
+        )
+        const id = new ZkIdentity()
+        await unirepSocialContract
+            .userSignUp(id.genIdentityCommitment())
+            .then((t) => t.wait())
+        const userState = await genUserState(
+            ethers.provider,
+            unirepContract.address,
+            id
+        )
+        const proveGraffiti = BigInt(0)
+        const minPosRep = 0
+        const graffitiPreImage = BigInt(0)
+        const epkNonce = 0
+        const defaultPostReputation = 5
+
+        const reputationProof = await userState.genProveReputationProof(
+            attesterId,
+            epkNonce,
+            minPosRep,
+            proveGraffiti,
+            graffitiPreImage,
+            defaultPostReputation
+        )
+        const isValid = await reputationProof.verify()
+        expect(isValid, 'Verify reputation proof off-chain failed').to.be.true
+
+        const isProofValid = await unirepContract.verifyReputation(
+            reputationProof.publicSignals,
+            reputationProof.proof
+        )
+
+        expect(isProofValid, 'proof is not valid').to.be.true
+        const tx = await unirepSocialContract.publishPost(
+            'some post text',
+            reputationProof.publicSignals,
+            reputationProof.proof,
+            { value: attestingFee }
+        )
+        const receipt = await tx.wait()
+        expect(receipt.status, 'Submit post failed').to.equal(1)
```

Comment:
should use a `await userState.waitForSync()` to sync the state before query db

Reply:
after adding `await userState.waitForSync()` what should the exact query db be? I have the post query commented out here with `hashedContent` but think `hashedContent` needs to be replaced
```
const post = await db.findOne('Post', { where:{hashedContent}})
expect(post.hashedContent === hashedContent)
```

Reply:
You can check if the post exists with the information before
We have `hashedContent` now, and also you can check transactionHash,... and so on
You can refer to the [`Synchronizer`](https://github.com/Unirep/Unirep-Social/blob/f7a366f13ce850f75d0567fb06756878c3469dea/packages/core/src/synchronizer.ts#L236) to see what can be queried.
e.g. when the post event emits, it stores
```typescript
            db.create('Post', {
                transactionHash,
                hashedContent,
                epochKey,
                epoch,
                proveMinRep: minRep !== 0 ? true : false,
                minRep,
                posRep: 0,
                negRep: 0,
                status: 1,
            })
``` 
These are all the variables you can check

Reply:
OK, I think I understand. But _before_ this code:
```js
const post = await db.findOne('Post', { where:{hashedContent}})
expect(post.hashedContent === hashedContent)
```
would I have to create the `hashedContent` and any other values that can be queried? 
Because if you add those two lines of code here, the test would fail.

Reply:
you should merge the `main` branch
and there is `hashedContent` before submitting posts
https://github.com/Unirep/Unirep-Social/blob/2f08db0e0a38b4fc9ff8c8c3a138a42be2c2bb4d/packages/core/test/Posting.ts#L121

Reply:
Got it. But what is the best way to actually create the Post in the test? Currently, the test would fail because the `post.hashedContent` is `null`.